packages/iov-cosmos/src/cosmosconnection.ts

```diff
@@ -191,7 +196,35 @@ export class CosmosConnection implements BlockchainConnection {
     const { txhash, raw_log } = await this.restClient.postTx(tx);
     const transactionId = txhash as TransactionId;
     const firstEvent: BlockInfo = { state: TransactionState.Pending };
-    const producer = new DefaultValueProducer<BlockInfo>(firstEvent);
+    let blockInfoInterval: NodeJS.Timeout;
+    let lastEventSent: BlockInfo;
+    const producer = new DefaultValueProducer<BlockInfo>(firstEvent, {
+      onStarted: () => {
+        blockInfoInterval = setInterval(async () => {
+          const search = await this.searchTx({ id: transactionId });
+          if (search.length) {
```

Comment:
I think this "first" implementation gives you better type safety:

```ts
const foundTransaction = (await this.searchTx({ id: transactionId })).find(() => true);
```

However, this is a matter of taste I guess.

Reply:
I was just copying the pattern from the Lisk connection but I agree this is nicer.